### ISSUE TITLE
Pre-fill sequence textarea on default page with POST (without databases[] or advanced)

### ISSUE THREAD

**Issue by sammyjava** [wurmlab/sequenceserver#491]
Pardon me if this is a repeat, but I've not found it anywhere. I'd like to simply send a sequence to the SequenceServer via GET or more likely POST and have the user then choose the databases, add any BLAST options, and submit the search. I've been able to do the full POST of method+sequence+databases[]+advanced, but it seems a bit silly for me to rebuild that form when all I really want to do is get a sequence from my webapp (InterMine) over to SequenceServer. Is there already a way to do this? Thanks!

**Comment by sammyjava** [comment]
A great way to implement this would be to make databases[] not required by the POST API (sequence and method could remain required, or maybe not even method). If databases[] is not supplied, display the search form with the sequence (and method) set from the POST vars.

**Comment by yeban** [comment]
If the query sequence is contained in the BLAST databases, you can pre-fill the sequence textarea using its id:

`/?query=sequence_id`
`/?query=sequence_id:start_cord-stop_coord`

If the above doesn't work for you, pre-filling on POST could also be implemented, but I would be more in the favour of using Javascript to open sequenceserver in a new window and paste in the query sequence. It is 2020, surely the latter should be possible in a cross-browser manner.

**Comment by sammyjava** [comment]
The query sequence is arbitrary. We're blasting it against the BLAST databases to see what we get. It's FASTA text.

I agree that Javascript could be used to launch a new window with the texteara pre-filled, but that requires that anyone that wishes this functionality needs to code it themselves, while a simple presentation of the search page with the textarea prefilled from a sequence POST works "out of the box" for the world. Sometimes server-side solutions are preferable, even in 2020.

**Comment by yeban** [comment]
I am a bit confused about what you mean by "out of the box". On the page that links to SequenceServer, you would implement a form that makes the post request, right?

Regarding the Javascript solution what I had in mind was to provide a source file that one could load on the page and activate with a URL, and HTML id of two elements, one to read the sequence from and one for the 'submit/blast' button. Something like:

```html
<script src="/sequenceserver.js"></script>
<script>
SequenceServer.sendTo('url', 'id_of_sequence_field', 'id_of_blast_button')
<script>
```

**Comment by sammyjava** [comment]
> I am a bit confused about what you mean by "out of the box". On the page that links to SequenceServer, you would implement a form that makes the post request, right?

"Out of the box" meaning functionality exactly as you have already implemented with the POST API, except not _requiring_ `databases[]`. If `databases[]` is missing from the POSTed data, respond with the pre-filled textarea on the search form rather than an error message.

I do POST the data from my site, but using a form with just a button and hidden fields containing the sequence and the method. This is what I want to be able to implement:
```
<form method="POST" action="https://legumeinfo.org/sequenceserver/">
<input type="hidden" name="sequence" value=">gene1234\nACGTACGTACGTACTGACGTACGTACGTACGT"/>
<input type="hidden" name="method" value="blastn"/>
<input type="submit" value="BLAST"/>
</form>
```
The key is that I don't need to supply `databases[]` as I currently do.

(Note that you already have a variable `input_sequence` but I can't seem to set it with a POST.)

Are you saying I can write a script to prefill a form on, say, Amazon? Because that's essentially what I need to do, except it's a SequenceServer somewhere. I'm pretty sure I need to send that data over with an HTTP request. I'm transferring data from the user's browser to a sequence server which is totally separate from the page the browser has loaded, so the user can continue by choosing the desired databases and executing the BLAST on that SequenceServer.

> Regarding the Javascript solution what I had in mind was to provide a source file that one could load on the page and activate with a URL, and HTML id of two elements, one to read the sequence from and one for the 'submit/blast' button. Something like:
> 
> ```
> <script src="/sequenceserver.js"></script>
> <script>
> SequenceServer.sendTo('url', 'id_of_sequence_field', 'id_of_blast_button')
> </script>
> ```

**Comment by sammyjava** [comment]
After a bit of research, I saw that you had the option to post `input_sequence` in `routes.rb`, but it was throwing an error. I updated/fixed it with PR #493 . Hope you're OK with that, it does what I want!!

**Comment by yeban** [comment]
Thanks. The feature was contributed by lepbase. I am glad you fixed it. It would currently have taken me a few days to get back to this issue. I will update my notes that we need tests for this feature so it doesn't break again.
